tests/phpunit/test-wc-stripe-helper.php

```diff
@@ -507,4 +507,42 @@ public function test_add_stripe_methods_in_woocommerce_gateway_order() {
 		$this->assertTrue( $gateway_order['stripe'] < $gateway_order['stripe_affirm'] );
 		$this->assertTrue( $gateway_order['stripe_affirm'] < $gateway_order['cheque'] );
 	}
+
+	/**
+	 * Test for `add_mandate_data`.
+	 *
+	 * @param string $index The index of the server variable to set.
+	 * @param string $value The value to set the server variable to.
+	 * @param string $expected The expected IP address.
+	 * @dataProvider provider_test_add_mandate_data
+	 * @return void
+	 */
+	public function test_add_mandate_data( $index, $value, $expected ) {
+		unset( $_SERVER['REMOTE_ADDR'] );
+		unset( $_SERVER['HTTP_X_REAL_IP'] );
+		unset( $_SERVER['HTTP_X_FORWARDED_FOR'] );
+
+		$_SERVER[ $index ] = $value;
```

Comment:
Non-blocking nit: the variable names `$index`, `$value`, and `$expected` are all super-generic. It would be nice to use more specific variable names so it's easier to grok what each value is doing and represents.
```suggestion
	/**
	 * Test for `add_mandate_data`.
	 *
	 * @param string $server_variable_key   The key of the server variable to set.
	 * @param string $server_variable_value The value to set the server variable to.
	 * @param string $expected_ip_address    The expected IP address.
	 * @dataProvider provider_test_add_mandate_data
	 * @return void
	 */
	public function test_add_mandate_data( $server_variable_key, $server_variable_value, $expected_ip_address ) {
		unset( $_SERVER['REMOTE_ADDR'] );
		unset( $_SERVER['HTTP_X_REAL_IP'] );
		unset( $_SERVER['HTTP_X_FORWARDED_FOR'] );

		$_SERVER[ $server_variable_key ] = $server_variable_value;
```